### PR TITLE
Adding reason For fail

### PR DESCRIPTION
**Pull request by sojanpr**
Adding the reason provided by brach servers for failure in case of any
request to server failed.

@Sarkar @Sarkar

### CODE REVIEW

**Comment by sojanpr** [comment]
Ready for review

File: Branch-SDK/src/io/branch/referral/ServerRequestApplyReferralCode.java [hunk 1]
```diff
@@ -83,9 +83,9 @@ public void onRequestSucceeded(ServerResponse resp, Branch branch) {
     }
 
     @Override
-    public void handleFailure(int statusCode) {
+    public void handleFailure(int statusCode, String causeMsg) {
         if (callback_ != null) {
-            callback_.onInitFinished(null, new BranchError("Trouble applying the referral code.", statusCode));
+            callback_.onInitFinished(null, new BranchError("Trouble applying the referral code."+causeMsg, statusCode));
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/ServerRequestApplyReferralCode.java]: add some spacing here

File: Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java [hunk 2]
```diff
@@ -132,10 +132,10 @@ public void onUrlAvailable(String url) {
     }
 
     @Override
-    public void handleFailure(int statusCode) {
+    public void handleFailure(int statusCode, String causeMsg) {
         if (callback_ != null) {
             String failedUrl = getLongUrl();
-            callback_.onLinkCreate(failedUrl, new BranchError("Trouble creating a URL.", statusCode));
+            callback_.onLinkCreate(failedUrl, new BranchError("Trouble creating a URL."+causeMsg, statusCode));
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/ServerRequestCreateUrl.java]: same here

**Comment by aaustin** [comment]
On all of the Log statements, the string spacing and variable spacing are a bit off. Please make sure it's like this: `" + erro`. Then also, if it's appending an error message to the existing String, can you make sure there's a space in the string: `error here. " + error`.

**Comment by sojanpr** [comment]
Added correction for format issues . Ready for review

**Comment by aaustin** [comment]
Looks good :+1:

**Pull request merged**